Fix availability versions for spring 2025 releases

emw-apple opened the pull request:
#### d4eda55d54c54e22acc370dabcd15be77df3e90a
<pre>
Fix availability versions for spring 2025 releases
<a href="https://bugs.webkit.org/show_bug.cgi?id=294299#">https://bugs.webkit.org/show_bug.cgi?id=294299#</a>
<a href="https://rdar.apple.com/144997613">rdar://144997613</a>

Reviewed by Timothy Hatcher and Alexey Proskuryakov.

Mechanically update availability versions for declarations that shipped
in the release.

These files had conflicts and were manually resolved:

    Source/WebKit/UIProcess/API/Cocoa/WKFrameInfoPrivate.h
    Source/WebKit/UIProcess/API/Cocoa/WKOpenPanelParameters.h
    Source/WebKit/UIProcess/API/Cocoa/WKOpenPanelParametersPrivate.h
    Source/WebKit/UIProcess/API/Cocoa/WKPreferencesPrivate.h
    Source/WebKit/UIProcess/API/Cocoa/WKUIDelegate.h
    Source/WebKit/UIProcess/API/Cocoa/WKUIDelegatePrivate.h
    Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfiguration.h
    Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h

* Source/WebKit/UIProcess/API/Cocoa/WKContentWorldPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKFrameInfoPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKNavigationAction.h:
* Source/WebKit/UIProcess/API/Cocoa/WKNavigationDelegatePrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKNavigationResponsePrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKPreferencesPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKProcessPoolPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKUIDelegatePrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKUserScriptPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtension.h:
(NS_ERROR_ENUM):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionAction.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionCommand.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionContext.h:
(NS_ERROR_ENUM):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionController.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionControllerConfiguration.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionControllerDelegate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionControllerDelegatePrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionDataRecord.h:
(NS_ERROR_ENUM):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionDataType.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionMatchPattern.h:
(NS_ERROR_ENUM):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionMessagePort.h:
(NS_ERROR_ENUM):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionPermission.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionPermissionPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionTab.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionTabConfiguration.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionWindow.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionWindowConfiguration.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfiguration.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfigurationPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivateForTesting.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStorePrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKAutomationSessionDelegate.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKContentWorldConfiguration.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKProcessPoolConfiguration.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKTextRun.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKWebAuthenticationPanel.h:

Canonical link: <a href="https://commits.webkit.org/296100@main">https://commits.webkit.org/296100@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/67fa75f956c024d046dc57b39b69c8b364c46d08

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/107287 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/26969 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/17373 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/112501 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/57821 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/27650 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/35469 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/81438 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/110216 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/21915 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/96698 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/61810 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/21344 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/57266 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/91274 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/14862 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/115602 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/34353 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/25313 "Passed tests") | [⏳ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/GTK-WK2-Tests-EWS "Waiting to run tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/34729 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/92948 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/90220 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23020 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/35123 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/12910 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/30083 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/34275 "Built successfully") | [❌ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/39802 "Failed to build and analyze WebKit") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/34021 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/37376 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/35682 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Source/WebKit/UIProcess/API/Cocoa/WKUIDelegate.h:
```diff
@@ -292,7 +292,7 @@ WK_SWIFT_UI_ACTOR
  the web view to act as if the user selected the Cancel button on iOS or visionOS, immediately call the completion handler with nil.
  */
-- (void)webView:(WKWebView *)webView runOpenPanelWithParameters:(WKOpenPanelParameters *)parameters initiatedByFrame:(WKFrameInfo *)frame completionHandler:(WK_SWIFT_UI_ACTOR void (^)(NSArray<NSURL *> * _Nullable URLs))completionHandler WK_API_AVAILABLE(macos(10.12), ios(18.4), visionos(2.4));
+- (void)webView:(WKWebView *)webView runOpenPanelWithParameters:(WKOpenPanelParameters *)parameters initiatedByFrame:(WKFrameInfo *)frame completionHandler:(WK_SWIFT_UI_ACTOR void (^)(NSArray<NSURL *> * _Nullable URLs))completionHandler WK_API_AVAILABLE(macos(10.12), ios(18.4));
```
Before I look at others, why is this the right thing to do? I'm also surprised at how it's so new on iOS while so old on macOS.

emw-apple replied:
Oh, this one does look like a mistake in conflict resolution. Lily changed it in 0c0fedc14f04e89d2ed7fb0d05d391ca6974fbef and I stomped over it here.

emw-apple replied:
And it seems like the newness on iOS is legitimate. https://commits.webkit.org/284471@main made it work on iOS family platforms.

emw-apple replied:
OK, updated the commit and manually reviewed the files that had conflicts.

emw-apple committed:
Fix availability versions for spring 2025 releases
https://bugs.webkit.org/show_bug.cgi?id=294299#
rdar://144997613

Reviewed by Timothy Hatcher and Alexey Proskuryakov.

Mechanically update availability versions for declarations that shipped
in the release.

These files had conflicts and were manually resolved:

    Source/WebKit/UIProcess/API/Cocoa/WKFrameInfoPrivate.h
    Source/WebKit/UIProcess/API/Cocoa/WKOpenPanelParameters.h
    Source/WebKit/UIProcess/API/Cocoa/WKOpenPanelParametersPrivate.h
    Source/WebKit/UIProcess/API/Cocoa/WKPreferencesPrivate.h
    Source/WebKit/UIProcess/API/Cocoa/WKUIDelegate.h
    Source/WebKit/UIProcess/API/Cocoa/WKUIDelegatePrivate.h
    Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfiguration.h
    Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h

* Source/WebKit/UIProcess/API/Cocoa/WKContentWorldPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKFrameInfoPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKNavigationAction.h:
* Source/WebKit/UIProcess/API/Cocoa/WKNavigationDelegatePrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKNavigationResponsePrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKPreferencesPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKProcessPoolPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKUIDelegatePrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKUserScriptPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtension.h:
(NS_ERROR_ENUM):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionAction.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionCommand.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionContext.h:
(NS_ERROR_ENUM):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionController.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionControllerConfiguration.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionControllerDelegate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionControllerDelegatePrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionDataRecord.h:
(NS_ERROR_ENUM):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionDataType.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionMatchPattern.h:
(NS_ERROR_ENUM):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionMessagePort.h:
(NS_ERROR_ENUM):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionPermission.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionPermissionPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionTab.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionTabConfiguration.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionWindow.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionWindowConfiguration.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfiguration.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfigurationPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivateForTesting.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStorePrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKAutomationSessionDelegate.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKContentWorldConfiguration.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKProcessPoolConfiguration.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKTextRun.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKWebAuthenticationPanel.h:

Canonical link: https://commits.webkit.org/296100@main
